database/ticket_test.go

```diff
@@ -332,3 +361,101 @@ func testCountTickets(t *testing.T) {
 
 	count("revoked", 1, 1, 1)
 }
+
+func tAltHist() *AltSigHistory {
+	return &AltSigHistory{
+		Addr: randString(35, addrCharset),
+		Req:  randBytes(1000),
+		Sig:  randBytes(72),
+	}
+}
+
+func testAddAltSigHistory(t *testing.T) {
+	ticket := exampleTicket()
+
+	// This should error if no ticket exists for hash in the database.
+	if err := db.AddAltSigHistory(ticket.Hash, tAltHist()); err == nil {
+		t.Fatal("expected error for no ticket in db")
+	}
+
+	// Insert a ticket into the database.
+	err := db.InsertNewTicket(ticket)
+	if err != nil {
+		t.Fatalf("unexpected error storing ticket in database: %v", err)
+	}
+
+	// First insert is fine.
+	if err := db.AddAltSigHistory(ticket.Hash, tAltHist()); err != nil {
+		t.Fatalf("unexpected error for first alt history: %v", err)
+	}
+
+	// Over the max, currently 1, should error.
+	if err := db.AddAltSigHistory(ticket.Hash, tAltHist()); err == nil {
+		t.Fatalf("expected error for second alt history: %v", err)
+	}
+	if err := db.AddAltSigHistory(ticket.Hash, tAltHist()); err == nil {
+		t.Fatalf("expected error for third alt history: %v", err)
+	}
+}
+
+func testAltSigHistory(t *testing.T) {
+	ticket := exampleTicket()
+
+	// No ticket means no history.
+	history, err := db.AltSigHistory(ticket.Hash)
+	if err != nil {
+		t.Fatalf("unexpected error for no ticket in db: %v", err)
```

Comment:
s / unexpected error for no ticket in db / unexpected error fetching alt signature history for ticket

Reply:
ok